src/keys/nibble_key.rs

```diff
@@ -152,6 +135,7 @@ impl std::ops::Index<std::ops::RangeFrom<usize>> for NibbleKey {
 impl std::ops::Index<std::ops::RangeTo<usize>> for NibbleKey {
     type Output = [u8];
 
+    // Problem here
```

Comment:
@Scaeff this is the second major problem: I would like to implement the iterator as just going over the key's bytes and extracting either the nibble or bit. But since I have to return `&Self::Output`, I find myself returning the whole byte instead of just the bit/nibble. Any idea?

Reply:
If you don't want `[u8]`, what happens if you set `Self::Output` to something else?

Reply:
I still need to pass a reference to something, and that means I would have to create it on the fly, which isn't possible as it would go out of scope. I could somehow try to create a temporary element in a `Vec` and return a ref to it as a slice, however I'm not sure exactly how it would work and also I would not returning a reference to the underlying structure, so I would require some form of allocation. Not a show stopper, but not great either.